### ISSUE TITLE
Cmake errors building PCL 1.8 with VTK 7.0.0

### ISSUE THREAD

**Issue by Chungzuwalla** [PointCloudLibrary/pcl#1530]
My target is Visual Studio 2015. I understand VTK OpenGL2 backend is not yet supported, so I switched VTK to use the OpenGL backend.  I can configure, generate, build and install VTK with no errors. However after configuring and generating with PCL:

- Undefined VTKRENDERINGOPENGL_EXPORT causes many errors in vtkgl.h. VTKRENDERINGOPENGL_EXPORT is defined in vtkRenderingOpenGLModule.h.  Including vtkRenderingOpenGLModule.h in vtkgl.h fixed these.

- "C:\Users\richard\Documents\UXT\trunk\third-party\pcl\visualization\src\vtk\vtkVertexBufferObject.cxx(104): error C2039: 'GetExtensionManager': is not a member of 'vtkOpenGLRenderWindow'
  C:\Program Files (x86)\VTK\include\vtk-7.0\vtkOpenGLRenderWindow.h(40): note: see declaration of 'vtkOpenGLRenderWindow'."
class vtkOpenGLRenderWindow is tagged with VTKRENDERINGOPENGL2_EXPORT which is suspicious.

- "pcl_visualization" project attempts to link with "vtkRenderingOpenGL2.lib" and "vtkDomainsChemistryOpenGL2.lib", which look like OpenGL2 libraries.  "vtkDomainsChemistryOpenGL2.lib" does not exist, causing a linker error. However, a library called "vtkDomainsChemistryOpenGL2-7.0.lib" exists, so it appears a version suffix is added when VTK is built, but is not added when CMake generates the list of libraries to link PCL projects with.  I don't know if the problem is in VTK or PCL build system.  I'm not familiar with CMake.

Thanks for any suggestions.

**Comment by VictorLamoine** [comment]
PCL does not support VTK compiled with OPENGL2 backend.
Please see: https://github.com/PointCloudLibrary/pcl/issues/1275

**Comment by VictorLamoine** [comment]
After changing VTK OpenGL2 backend to OpenGL, please uninstall VTK then compile it again and install it.

You also need to completely reconfigure PCL otherwise there are some clashes when you switch from OpenGL2 to OpenGL.

Try to reload the re-create the CMake cache and compile again.

**Comment by UnaNancyOwen** [comment]
I was able to build PCL 1.8.0rc1 (Boost 1.60.0, Eigen 3.3.0 beta1, FLANN 1.8.4, QHull 2015.2, VTK 7.0.0, OpenNI 2.2.0.33 Beta) with Visual Studio 2015.
My build-notes might become your reference. (Sorry, This build-notes is written in Japanese.)
Also, I have distribute the All-in-one Installer.

* Build-Notes https://gist.github.com/UnaNancyOwen/59319050d53c137ca8f3#file-pcl1-8-0-md
* All-in-one Installer http://unanancyowen.com/?p=1794&lang=en

I hope it can be of any help to you.

**Comment by VictorLamoine** [comment]
@UnaNancyOwen  With VTK OpenGL2 back-end? Or OpenGL back-end?

**Comment by UnaNancyOwen** [comment]
@VictorLamoine No, It is OpenGL back-end :(
Sorry, I have no enough knowledge of VTK for can be solved that problem.

**Comment by Chungzuwalla** [comment]
@UnaNancyOwen I found your excellent all-in-one builds for VS2015 when I first started using PCL.  I think they would be very useful for me.  Unfortunately I don’t work at a very forward thinking company, so access to download your installers from onedrive.live.com is blocked.  Even more unfortunately, it is a Japanese company. :-(
Thanks anyway for your help.

@VictorLamoine I confirmed that when I built the new backend, OpenGL2 modules were left in the VTK install directory. VTKConfig.cmake attempts to link projects against all VTK modules that it finds in the install directory, including both incompatible backends. So I had to manually delete the VTK install directory to solve the issue. I think that is probably a problem with the VTK build system (it should at least issue a warning), so I've transferred the issue to the VTK mailing list.  Thanks for your help.
